src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -121,23 +122,50 @@ public DateTime GetUpdatedAt(Document document, List<GitCommit> fileCommits)
             if (repo == null)
                 return default;
 
-            var branch = repo.Branch ?? "master";
-            var editRepo = LocConfigConvention.GetEditRepository(document.Docset.Config.Contribution.Repository, document.Docset.Locale, document.Docset.Config.DefaultLocale)
-                ?? $"{repo.Owner}/{repo.Name}";
-            var editBranch = document.Docset.Config.Contribution.Branch ?? branch;
-
-            var editUrl = document.Docset.Config.Contribution.ShowEdit
-                ? $"https://github.com/{editRepo}/blob/{editBranch}/{pathToRepo}"
-                : null;
-
+            var repoHost = GithubUtility.TryParse(repo.Remote, out _, out _) ? GitHost.GitHub : GitHost.Unknown;
             var commit = _commitsByFile.TryGetValue(document.FilePath, out var value) && value.commits.Count > 0
                 ? value.commits[0].Sha
                 : repo.Commit;
 
-            var commitUrl = commit != null ? $"https://github.com/{repo.Owner}/{repo.Name}/blob/{commit}/{pathToRepo}" : null;
-            var contentUrl = $"https://github.com/{repo.Owner}/{repo.Name}/blob/{branch}/{pathToRepo}";
+            return (GetEditUrl(), GetContentUrl(), GetCommitUrl());
+
+            string GetCommitUrl()
+            {
+                switch (repoHost)
+                {
+                    case GitHost.GitHub:
+                        return commit != null ? $"{repo.Remote}/blob/{commit}/{pathToRepo}" : null;
+                    default:
+                        throw new NotSupportedException($"{repoHost}");
+                }
+            }
 
-            return (editUrl, contentUrl, commitUrl);
+            string GetContentUrl()
+            {
+                switch (repoHost)
+                {
+                    case GitHost.GitHub:
+                        return $"{repo.Remote}/blob/{repo.Branch}/{pathToRepo}";
+                    default:
+                        throw new NotSupportedException($"{repoHost}");
+                }
+            }
+
+            string GetEditUrl()
+            {
+                // git edit url, only works for github repo
```

Comment:
> // git edit url, only works for github repo [](start = 15, length = 44)

it will also work for vsts when it's supported. Also use switch + throw NotSupportedException?

Reply:
do we really have editable VSTS links? VSTS repos are all private and requires authentication, even if we generate a link, users won't have access to it.

Reply:
VSTS is all private repo, how to enable contribution?

---
In reply to: [224650129](https://github.com/dotnet/docfx/pull/3485#discussion_r224650129) [](ancestors = 224650129)

Reply:
agree

---
In reply to: [224650639](https://github.com/dotnet/docfx/pull/3485#discussion_r224650639) [](ancestors = 224650639)

Reply:
I just leave this as by design, in the feature we can change it if we want to support it.

---
In reply to: [224651493](https://github.com/dotnet/docfx/pull/3485#discussion_r224651493) [](ancestors = 224651493,224650129)